ArcGISRuntimeSDKQt_CppSamples/Analysis/DisplayUtilityAssociations/DisplayUtilityAssociations.cpp

```diff
@@ -0,0 +1,275 @@
+      {
+        edges.append(networkSource);
+      }
+      else if (networkSource->sourceType() == UtilityNetworkSourceType::Junction)
+      {
+        junctions.append(networkSource);
+      }
+    }
+
+    // add all edges that are not subnet lines to the map
+    for (UtilityNetworkSource* source : edges)
+    {
+      if (source->sourceUsageType() != UtilityNetworkSourceUsageType::SubnetLine && source->featureTable() != nullptr)
+      {
+        m_map->operationalLayers()->append(new FeatureLayer(source->featureTable(), this));
+      }
+    }
+
+    // add all junctions to the map
+    for (UtilityNetworkSource* source : junctions)
+    {
+      if (source->featureTable())
+      {
+        m_map->operationalLayers()->append(new FeatureLayer(source->featureTable(), this));
+      }
+    }
+
+    // create a renderer for the associations
+    UniqueValue* attachmentValue = new UniqueValue("Attachment", "",
+                                                   QVariantList{static_cast<int>(UtilityAssociationType::Attachment)}, m_attachmentSymbol, this);
+    UniqueValue* connectivityValue = new UniqueValue("Connectivity", "",
+                                                     QVariantList{static_cast<int>(UtilityAssociationType::Connectivity)}, m_connectivitySymbol, this);
+    UniqueValueRenderer* uniqueValueRenderer = new UniqueValueRenderer("", nullptr,
+                                                                       QStringList{"AssociationType"}, QList<UniqueValue*>{attachmentValue, connectivityValue}, this);
+    m_associationsOverlay->setRenderer(uniqueValueRenderer);
+
+    // populate the legend
+    m_attachmentSymbol->createSwatch();
+    m_connectivitySymbol->createSwatch();
+  });
+
+  connect(m_utilityNetwork, &UtilityNetwork::associationsCompleted, this, [this](QUuid, const QList<UtilityAssociation*>& associations)
+  {
+    bool matchedGraphic = false;
+    for (UtilityAssociation* association : associations)
+    {
+      matchedGraphic = false;
+      // check if the graphics overlay already contains the association
+      for (Graphic* graphic : *m_associationsOverlay->graphics())
+      {
+        if (graphic->attributes()->containsAttribute("GlobalId") && qvariant_cast<QUuid>(graphic->attributes()->operator[]("GlobalId")) == association->globalId())
+        {
+          matchedGraphic = true;
+          break;
+        }
+      }
+
+      // if the association is already in the overlay, don't add a new graphic
+      if (matchedGraphic)
+      {
+        continue;
+      }
+
+      // add a graphic for the association
+      QVariantMap graphicAttributes;
+      graphicAttributes["GlobalId"] = association->globalId();
+      graphicAttributes["AssociationType"] = static_cast<int>(association->associationType());
+      Graphic* graphic = new Graphic(association->geometry(), graphicAttributes, this);
+
+      m_associationsOverlay->graphics()->append(graphic);
+    }
```

Comment:
We can remove `matchedGraphic` and these break/continues. This is a higher order function at its core.

```cpp
const auto& graphics = *m_associationsOverlay->graphics();

const bool unqiueGraphic = std::none_of(graphics.begin(), graphics.end(), [=](const Graphic* graphic) { 
    const auto& attributes = *graphic->attributes();
    return attributes.containsAttribute("GlobalId") &&
            graphic->attributes["GlobalId"].toUuid() == association->globalId();
});

if (unqiueGraphic)
{
    QVariantMap graphicAttributes;
    graphicAttributes["GlobalId"] = association->globalId();
    graphicAttributes["AssociationType"] = association->associationType().toInt();
    Graphic* graphic = new Graphic(association->geometry(), graphicAttributes, this);
}

```

Reply:
@anmacdonald thanks that's much more readable!